facebook/delphiFacebook/R/responses.R

```diff
@@ -18,13 +18,29 @@
 #' @export
 load_responses_all <- function(params, contingency_run = FALSE) {
   msg_plain(paste0("Loading ", length(params$input), " CSVs"))
-  
-  map_fn <- if (params$parallel) { mclapply } else { lapply }
+
+  # To prevent errors from one file from "contaminating" output for other files
+  # that happen to be in the same `mclapply` group, turn `mc.preschedule` off.
+  # This makes debugging problematic input files easier.
+  # See https://stackoverflow.com/questions/18330274/an-error-in-one-job-contaminates-others-with-mclapply
+  # for details.
+  mclapply_no_preschedule <- function(...) {
+    mclapply(..., mc.preschedule = FALSE)
```

Comment:
Hmm. With prescheduling turned off, `mclapply` will apparently fork one child process per data file to read (though only at most `mc.cores` at a time). That means there will be m ore time spent starting up new processes. Did you try this out on any test data to see if it has a significant performance cost?

I'm hoping R's startup time is negligible compared to the time of reading in big huge CSVs, but I dunno.

Reply:
Good question, I'll run some tests.

Reply:
Total CPU time of loading 46 input files 427m with prescheduling vs 467m without prescheduling. So loading input files in parallel is ~9% slower without prescheduling. That's not insignificant, so I'll switch prescheduling back on and make the error message clearer.